ui/analyse/src/ctrl.ts

```diff
@@ -468,6 +473,38 @@ export default class AnalyseCtrl {
       '/' +
       encodeURIComponent(fen).replace(/%20/g, '_').replace(/%2F/g, '/');
   }
+  returnStandardCastlingFen(fen: Fen): Fen {
+    let new_fen = fen;
+    const setup = parseFen(fen).unwrap();
+    let castlingPart = '-';
+    const iterable = setup.unmovedRooks[Symbol.iterator]();
+    const w_castling_squares = [];
+    const b_castling_squares = [];
+    const kings_in_starting = {
+      //check if white king on its starting square
+      w: setup.board.king.has(4) && setup.board.white.has(4),
+      //check same thing for black king
+      b: setup.board.king.has(60) && setup.board.black.has(60),
+    };
+    let result = iterable.next();
+    //populate arrays to contain possible black and white castling squares
+    while (!result.done) {
+      if (setup.board.black.has(result.value)) b_castling_squares.push(result.value);
+      else if (setup.board.white.has(result.value)) w_castling_squares.push(result.value);
+      result = iterable.next();
+    }
+    //generate new fen if necessary
+    if ((kings_in_starting['w'] && w_castling_squares) || (kings_in_starting['b'] && b_castling_squares)) {
+      castlingPart =
+        'K'.repeat(w_castling_squares.includes(7) ? 1 : 0) +
+        'Q'.repeat(w_castling_squares.includes(0) ? 1 : 0) +
+        'k'.repeat(b_castling_squares.includes(63) ? 1 : 0) +
+        'q'.repeat(b_castling_squares.includes(56) ? 1 : 0);
+    }
+    setup.unmovedRooks = parseCastlingFen(setup.board, castlingPart).unwrap();
+    new_fen = makeFen(setup);
+    return new_fen;
+  }
```

Comment:
yikes. That code certainly deserves its own file and tests. Should probably be in chessops too, as you mentioned.